### DISCUSSION TITLE
Help with a code in AlaSQL

### DISCUSSION THREAD

**Discussion by mnlabrana** [agershun/alasql#1907] [General]
Hi guys I have this code:

```
SELECT
   T1."T",
   CASE WHEN T1."Count 2024 Q1" IS NOT NULL THEN T1."Count 2024 Q1" END AS "Q1",
   CASE WHEN T1."Count 2024 Q2" IS NOT NULL THEN T1."Count 2024 Q2" END AS "Q2",
   CASE WHEN T1."Count 2024 Q3" IS NOT NULL THEN T1."Count 2024 Q3" END AS "Q3",
   CASE WHEN T1."Count 2024 Q4" IS NOT NULL THEN T1."Count 2024 Q4" END AS "Q4"
FROM T1
```

It is a table of 5 columns and 3 rows.
I would like create a new column and in the last 2 rows put a text for example:

New column:
Row 1 -> Tittle Column
Row 2 -> text A
Row 3 -> text B

so finally the main table (T1) will be with 6 columns and 3 rows. Do you know if is possible to make this in AlaSQL?

**Comment by mathiasrw** [comment]
_From the [gpt-4 bot](https://chat.openai.com/g/g-XcBL24WTe-alasql-bot)_

Yes, you can create a new column in AlaSQL and populate it with specific text values for certain rows. Here's how you could do it:

1. Add a new column in your select statement.
2. Use `CASE` or a similar conditional logic to specify the values for the new column based on the row number or other unique identifiers.

Here is the modified SQL statement based on your requirements:

```sql
SELECT
   T1."T",
   CASE WHEN T1."Count 2024 Q1" IS NOT NULL THEN T1."Count 2024 Q1" END AS "Q1",
   CASE WHEN T1."Count 2024 Q2" IS NOT NULL THEN T1."Count 2024 Q2" END AS "Q2",
   CASE WHEN T1."Count 2024 Q3" IS NOT NULL THEN T1."Count 2024 Q3" END AS "Q3",
   CASE WHEN T1."Count 2024 Q4" IS NOT NULL THEN T1."Count 2024 Q4" END AS "Q4",
   CASE 
       WHEN ROW_NUMBER() OVER (ORDER BY (SELECT NULL)) = 1 THEN 'Tittle Column' 
       WHEN ROW_NUMBER() OVER (ORDER BY (SELECT NULL)) = 2 THEN 'text A' 
       WHEN ROW_NUMBER() OVER (ORDER BY (SELECT NULL)) = 3 THEN 'text B' 
   END AS "New Column"
FROM T1
```

In this SQL, `ROW_NUMBER() OVER (ORDER BY (SELECT NULL))` is used to generate a sequence number for each row to determine which row to assign 'text A' and 'text B'. Adjust the `ORDER BY` clause as per your table's logical order.

This solution assumes that your table `T1` has a predictable sort order or that the row number corresponds to the actual rows where you want to add 'text A' and 'text B'. If the table's row order is not guaranteed, you might need a more reliable method to target specific rows, possibly using an identifier or key column in the table.

**Comment by singh0130** [comment]
Hi @mathiasrw,

I am new to this project and would really love to contribute here but i am really finding it difficult to how to start. Can you help me if there are any resources for how can i run the project and test how this thing works?

sorry for asking here :)
